Hardit_Singh/bankAccount.rb

```diff
@@ -0,0 +1,59 @@
+class BankAccount
+	attr_reader	:accountNumber,
+				:userName,
+				:checkingBalance,
+				:savingBalance
+
+	def initialize(userName,balance=0)		
+		@accountNumber = genAccountNumber
+		@userName = userName
+		@checkingBalance = balance
+		@savingBalance = balance
+		@@noOfAccounts += 1
+		self
+	end
+
+	def get(req)
+		eval"@#{req}"
+	end
+
+	def deposit(amount,acctype)
+		(['checking','saving'].include? acctype) ?  (eval"@#{acctype}Balance+=#{amount}") : 'Not a valid account type'
+		self
+	end
+
+	def total
+		puts "Total Balance: #{@checkingBalance+@savingBalance}"
+	end
+
+	def withdraw(amount,acctype)
+		# i know this is evil but just wanted to see how nested ternary operators work!
+		(!['checking','saving'].include? acctype) ?  "wrong account type" : (eval"@#{acctype}Balance>=#{amount}") ? (eval"@#{acctype}Balance-=#{amount}") : "Insufficient Funds" 

```

Comment:
A nested ternary is the evil of all evils, now that you see that I expect you to never do it again.